test/MaticTokenAirdrop.test.js

```diff
@@ -0,0 +1,134 @@
+const crypto = require('crypto')
+const bluebird = require('bluebird')
+const abiDecoder = require('abi-decoder')
+const parseCsv = require('./utils').parseCsv
+
+const Token = artifacts.require("./contracts/MaticToken.sol")
+const Airdrop = artifacts.require("./contracts/MaticTokenAirdrop.sol")
+const BigNumber = web3.utils.BigNumber
+
+abiDecoder.addABI(Token._json.abi);
+
+require("chai")
+  .use(require("chai-bignumber")(BigNumber))
+  .should()
+
+contract.only("MaticTokenAirdrop", async function(accounts) {
+  const SCALING_FACTOR = web3.utils.toBN(10 ** 18)
+  let owner = accounts[0]
+  let token
+  let airdrop
+  let totalSupply = web3.utils.toBN(10 ** 10).mul(SCALING_FACTOR)
+
+  beforeEach(async function() {
+    token = await Token.new("Matic test", "MATIC", 18, totalSupply, {from: owner})
+    airdrop = await Airdrop.new(token.address, { from: owner })
+  })
+
+  it("should airdrop from test data", async function() {
+    const NUM_RECIPIENTS = 200
+    let { recipients, amounts, airdropSupply } = buildTestData(NUM_RECIPIENTS)
+    // console.log(recipients, amounts, airdropSupply.toString())
+    airdropSupply = web3.utils.toBN(airdropSupply).mul(SCALING_FACTOR)
+
+    // transfer tokens to the contract for airdrop
+    await token.transfer(airdrop.address, airdropSupply, { from: owner })
+    let airdropContractBalance = await token.balanceOf.call(airdrop.address)
+    assert.ok(airdropContractBalance.eq(airdropSupply), 'airdropSupply assertion failed')
+
+    // assert initial user balances are 0
+    await bluebird.map(recipients, async recipient => {
+      let balance = await token.balanceOf.call(recipient)
```

Comment:
what's the point of using bluebird here when we are using await for each balance of call? await all would be better?

Reply:
`bluebird.map` will fire all `balanceOf` calls simultaneously. `await` will cause _each_ call to wait on the result before asserting its result 😃 But other calls will continue execution.